web/web/models/user.py

```diff
@@ -26,28 +26,51 @@ class User(UserMixin, Model):
                 nullable=False)
 
     # User email information
-    email = Column(db.String(255), nullable=False, unique=True)
+    email = Column(db.String(255),
+                   unique=True,
+                   nullable=False)
+
     email_confirmed_at = Column(TIMESTAMP)
 
     # User information
-    first_name = Column(db.String(64), nullable=False)
-    last_name = Column(db.String(64), nullable=False)
-    address_id = Column(db.Integer, db.ForeignKey('addresses.address_id'))
+    first_name = Column(db.String(64),
+                        nullable=False)
+
+    last_name = Column(db.String(64),
+                       nullable=False)
+
+    address_id = Column(db.Integer,
+                        db.ForeignKey('addresses.address_id'),
+                        nullable=False)
+
     utility_id = Column(db.Integer,
                         db.ForeignKey('utilities.utility_id'),
                         nullable=False)
 
```

Comment:
do you need utility and address ids if you are defining the backref below?

Reply:
Both are needed to created the relationship from my understanding. Ideally, the backref should be on the foreign key tables (such as address or utility), but there is a problem with circular imports that prevents this. I haven't been able to find a good solution for this issue.

Reply:
IAW the docs here: https://docs.sqlalchemy.org/en/13/orm/backref.html it seems like it shouldn't be required. Can you give a try and let me know if that works and maybe even solves the circular import issue.

Reply:
Update: I tried to create the database without the foreign key ids to run through it but it errors out needing a foreign key to create the join. Error: "Could not determine join condition between parent/child tables on relationship User.utility - there are no foreign keys linking these tables.  Ensure that referencing columns are associated with a ForeignKey or ForeignKeyConstraint, or specify a 'primaryjoin' expression". So, the foreign key is needed, but the backref should be on the address and utility tables instead of the user table (which runs into the circular import issue). I will work on trying to get backref on the address and utility tables instead of on User.

Reply:
I believe I fixed the issue! - declaring for example Utility.user on models/user.py 

Utility.user = relationship('User',
                            backref=db.backref('utilities'),
                            uselist=False)

Reply:
Nice, excited to see what shakes out of this.

However, what's the association between `user` and `utility`? Or, why does a `user` have a foreign key to `utility`?

Can/should one user be associated with more than one utility?